### PR TITLE
FUSETOOLS2-2075: UI test Jbang commands with automatic reload

### PR DESCRIPTION
**Pull request by unsortedhashsets**
Related:

https://issues.redhat.com/browse/FUSETOOLS2-2075

### CODE REVIEW

File: src/ui-test/tests/reload.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+describe('Jbang commands with automatic reload', function () {
+    this.timeout(240000);
+
+    let driver: WebDriver;
+    let resourceManager: IResourceManager;
+
+    before('Before setup', async function () {
+        driver = VSBrowser.instance.driver;
+
+        resourceManager = createResourceManager(VSBrowser.instance, createWorkspace(VSBrowser.instance, 'src/ui-test/resources'), 'src/ui-test/resources');
```
Comment (unsortedhashsets) [inline src/ui-test/tests/reload.test.ts:39]: @mlorinc 
Hi, thanks for advice according that Class
works nice do copy and delete

But I faced few issues:
1) by default it creates new folders in resources: `store` and `data` -> [gitlink](https://github.com/djelinek/vscode-uitests-tooling/blob/main/src/utils/Resources.ts#L96-L98)
2) by default `copy` use  `data` folder -> [gitlink](https://github.com/djelinek/vscode-uitests-tooling/blob/main/src/utils/Resources.ts#L125C47-L127) 

So I used `../${CAMEL_ROUTE_YAML_WITH_SPACE}` to copy/delete file
and also make extra `deleteFromStore`/`deleteStore` to delete folders due to the fact that `cleanup` deletes the whole resources folder

Comment (unsortedhashsets) [reply src/ui-test/tests/reload.test.ts:39]: Can it be issued in `vscode-uitests-tooling` to add more flexibility? Or we will adapt our folder structure to it?
WDYT? @mlorinc @djelinek

Comment (unsortedhashsets) [reply src/ui-test/tests/reload.test.ts:39]: As we discussed with @djelinek I will update `vscode-uitests-tooling`:
1) Do not create folders by default
2) By default copy from path should be `resources` not `data`
3) That PR is blocked due to that issue

File: src/ui-test/tests/reload.test.ts [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    killTerminal,
+    HELLO_CAMEL_MESSAGE,
+    activateTerminalView,
+    CAMEL_ROUTE_YAML_WITH_SPACE,
+    HELLO_WORLD_MESSAGE,
+    CAMEL_ROUTE_YAML_WITH_SPACE_COPY,
+    replaceTextInCodeEditor
+} from '../utils';
+import { expect } from 'chai';
+import { createResourceManager, IResourceManager } from 'vscode-uitests-tooling/out/utils/Resources';
```
Comment (djelinek) [inline src/ui-test/tests/reload.test.ts:27]: these imports cannot be imported directly from `vscode-uitests-tooling` ?

File: src/ui-test/tests/reload.test.ts (hunk 1 above)
Comment (unsortedhashsets) [reply src/ui-test/tests/reload.test.ts]: Ok, seems done

File: src/ui-test/tests/reload.test.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+    afterEach('After each test', async function () {
+        await killTerminal();
+    });
+
+    it(`Execute command '${CAMEL_RUN_ACTION_LABEL}' with automatic reload`, async function () {
+        await executeCommand(CAMEL_RUN_ACTION_LABEL);
+        await waitUntilTerminalHasText(driver, TEST_ARRAY_RUN);
+
+        await driver.wait(async () => { return await replaceTextInCodeEditor(HELLO_CAMEL_MESSAGE, HELLO_WORLD_MESSAGE); });
```
[the remark this answers is not in the document]
Comment (djelinek) [reply src/ui-test/tests/reload.test.ts:73]: do not forget to open JIRA for it, I would like to have it tracked and also I am curious what is it about

**Review by djelinek — APPROVED** [review]

File: src/ui-test/tests/reload.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (unsortedhashsets) [reply src/ui-test/tests/reload.test.ts:73]: FYI, it is just random notifications specific to windows
 
Screenshot from 2023-05-31 13-18-13

**Pull request merged**